module/zfs/spa_stats.c

```diff
@@ -997,6 +998,82 @@ spa_mmp_history_add(spa_t *spa, uint64_t txg, uint64_t timestamp,
 	return ((void *)smh);
 }
 
+static void *
+spa_health_addr(kstat_t *ksp, loff_t n)
+{
+	return (ksp->ks_private);	/* return the spa_t */
+}
+
+/*
+ * Return true if we should print "SUSPENDED" for the pool health.  This is the
+ * kernel-side version of zpool_suspended_no_continue().
+ */
+static boolean_t
+spa_suspended_no_continue(spa_t *spa)
+{
+	return (spa_suspended(spa) && (spa_get_failmode(spa)
+	    != ZIO_FAILURE_MODE_CONTINUE));
+}
```

Comment:
After adding `spa_state_to_name` you might find it clearer to move this check and the associated `strlcpy` in to the new function.    Then you wouldn't need a conditional at all in `spa_health_data`.